modules/history/module.py

```
@@ -75,27 +75,44 @@ def _get_ui_path(self, relative_path: str) -> Path:
+                return datetime.min
+
+        entries.sort(key=_sort_key, reverse=True)
+
+        for kind, entry in entries:
             row = self.history_table.rowCount()
             self.history_table.insertRow(row)
 
             try:
-                date = datetime.fromisoformat(job['date']).strftime("%Y-%m-%d %H:%M")
+                date = datetime.fromisoformat(entry['date']).strftime("%Y-%m-%d %H:%M")
             except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)